backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/tools/SpdxConnector.java

```diff
@@ -0,0 +1,117 @@
+import org.spdx.rdfparser.license.LicenseInfoFactory;
+import org.spdx.rdfparser.license.SpdxListedLicense;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class SpdxConnector {
+
+    private static final Logger log = Logger.getLogger(SpdxConnector.class);
+
+    public static List<String> getAllSpdxLicenseIds() {
+        return Arrays.asList(LicenseInfoFactory.getSpdxListedLicenseIds());
+    }
+
+    protected static Optional<SpdxListedLicense> getSpdxLicense(String licenseId) {
+        final SpdxListedLicense listedLicenseById;
+        try {
+            listedLicenseById = LicenseInfoFactory.getListedLicenseById(licenseId);
+        } catch (InvalidSPDXAnalysisException e) {
+            log.warn("Failed to find SpdxListedLicense with id: " + licenseId);
+            return Optional.empty();
+        }
+        return Optional.of(listedLicenseById);
+    }
+
+    public static Optional<License> getSpdxLicenseAsSW360License(String licenseId){
+        return getSpdxLicense(licenseId)
+                .flatMap(SpdxConnector::getSpdxLicenseAsSW360License);
+    }
+
+    public static Optional<License> getSpdxLicenseAsSW360License(SpdxListedLicense spdxListedLicense){
+        License license = new License()
+                .setId(spdxListedLicense.getLicenseId())
+                .setShortname(spdxListedLicense.getLicenseId())
+                .setFullname(spdxListedLicense.getName())
+                .setText(spdxListedLicense.getLicenseText())
+                .setExternalLicenseLink("https://spdx.org/licenses/" + spdxListedLicense.getLicenseId()+ ".html");
+        return Optional.of(license);
+    }
+
+    /**
+     * checks whether the given license matches the corresponding SPDX license
+     * @param license
+     * @return
+     */
+    public static boolean matchesSpdxLicenseText(License license) {
+        return matchesSpdxLicenseText(license, license.getShortname());
+    }
+
+    public static boolean matchesSpdxLicenseText(License license, String spdxId) {
+        final Optional<SpdxListedLicense> spdxLicense = getSpdxLicense(spdxId);
+        if(!spdxLicense.isPresent()) {
+            return true;
+        }
+        return matchesSpdxLicenseText(license, spdxLicense.get());
+    }
+
+    private static boolean matchesSpdxLicenseText(License license, SpdxListedLicense spdxLicense) {
+        return matchesLicenseText(license, spdxLicense.getLicenseText());
+    }
+
+    public static boolean matchesLicenseText(License license, String licenseText) {
+        return  LicenseCompareHelper.isLicenseTextEquivalent(license.getText(), licenseText);
+    }
+
+    /**
+     * returns a list of potentially matching SPDX license IDs
+     * @param license
+     * @return
+     * @throws SW360Exception
+     */
+    public static List<String> findMatchingSpdxLicenseIDs(License license) throws SW360Exception {
```

Comment:
These functions were implemented for a proof-of-concept and might be used in another feature.